### PR TITLE
api link

### LINKED ISSUE #241: Add an 'open in new tab' button to the documentation panel
An icon like the rectangle with an arrow emerging from it, at the top right hand side of the doc panel that enabled you to open the current documentation in a new tab, potentially via a link to api.dartlang.org

@kasperpeulen

### PR DESCRIPTION
**Pull request by kasperpeulen**
fix #241

### CODE REVIEW

File: lib/playground.dart [hunk 1]
```diff
@@ -400,31 +400,50 @@ class Playground {
...
 '''
-# `${result.info['description']}`\n\n
-${result.info['dartdoc'] != null ? result.info['dartdoc'] + "\n\n" : ""}
-${result.info['kind'].contains("variable") ? "${result.info['kind']}\n\n" : ""}
-${result.info['kind'].contains("variable") ? "**Propagated type:** ${result.info["propagatedType"]}\n\n" : ""}
-${result.info['libraryName'] != null ? "**Library:** ${result.info['libraryName']}" : ""}\n\n
+# `${info['description']}`\n\n
+${info['dartdoc'] != null ? info['dartdoc'] + "\n\n" : ""}
+${info['kind'].contains("variable") ? "${info['kind']}\n\n" : ""}
+${info['kind'].contains("variable") ? "**Propagated type:** ${result.info["propagatedType"]}\n\n" : ""}
```
Comment (lukechurch) [inline lib/playground.dart]: s/ ${result.info["propagatedType"]/ ${info["propagatedType"]/

File: lib/playground.dart [hunk 2]
```diff
@@ -400,31 +400,50 @@ class Playground {
...
           _docPanel.setInnerHtml(markdown.markdownToHtml(
 '''
-# `${result.info['description']}`\n\n
-${result.info['dartdoc'] != null ? result.info['dartdoc'] + "\n\n" : ""}
-${result.info['kind'].contains("variable") ? "${result.info['kind']}\n\n" : ""}
-${result.info['kind'].contains("variable") ? "**Propagated type:** ${result.info["propagatedType"]}\n\n" : ""}
-${result.info['libraryName'] != null ? "**Library:** ${result.info['libraryName']}" : ""}\n\n
+# `${info['description']}`\n\n
+${info['dartdoc'] != null ? info['dartdoc'] + "\n\n" : ""}
+${info['kind'].contains("variable") ? "${info['kind']}\n\n" : ""}
```
Comment (lukechurch) [inline lib/playground.dart]: factor info['kind'] into a variable?

File: lib/playground.dart [hunk 3]
```diff
@@ -400,31 +400,50 @@ class Playground {
...
+${info['kind'].contains("variable") ? "**Propagated type:** ${result.info["propagatedType"]}\n\n" : ""}
+${info['libraryName'] != null ? "**Library:** [${info['libraryName']}](${apiLink})" : ""}\n\n
```
Comment (lukechurch) [inline lib/playground.dart]: If the apiLink does return null - this is going to look a bit strange. Possible it needs to be conditional?

File: lib/services/common.dart [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 library dart_pad.common;
 
-final String serverURL = 'https://dart-services.appspot.com/';
+final String serverURL = 'http://doc-test.dart-services.appspot.com/';
```
Comment (lukechurch) [inline lib/services/common.dart]: We can't do this in the main application code, sorry.

We can either use a cond-flag to turn this behaviour on, or we can wait for a day or so until I've finished stress testing the doc-test API and can make the version mainline.

File: lib/playground.dart (hunk 3 above)
Comment (kasperpeulen) [inline lib/playground.dart]: oh you are right !

File: lib/playground.dart (hunk 2 above)
Comment (kasperpeulen) [inline lib/playground.dart]: yeah, I was tired of all those result.info's :P

File: lib/services/common.dart (hunk 4 above)
Comment (kasperpeulen) [inline lib/services/common.dart]: oh wait, forgot about that, probably we should wait for a day then

File: lib/playground.dart [hunk 5]
```diff
@@ -400,31 +400,51 @@ class Playground {
...
               => a.target = "_blank");
           _docPanel.querySelectorAll("h1").forEach((h)
-              => h.classes.add("type-${result.info["kind"].replaceAll(" ","_")}"));
+              => h.classes.add("type-${kind.replaceAll(" ","_")}"));
         }
       });
     }
   }
 
+  String _dartApiLink(Map info){
```
Comment (devoncarew) [inline lib/playground.dart]: Could you refactor this into taking 3 (probably named) params? Something like:
```
String _createApiLink({String libraryName, String enclosingClassName, String name}) { ...
```
It'll be clearer what the inputs of the method are.

File: lib/playground.dart [hunk 6]
```diff
@@ -400,31 +400,51 @@ class Playground {
...
+    StringBuffer apiLink = new StringBuffer();
+    if (info["libraryName"] != null) {
+      if (info["libraryName"].contains("dart:")) {
+        apiLink.write( "https://api.dartlang.org/apidocs/channels/stable/dartdoc-viewer/");
```
Comment (devoncarew) [inline lib/playground.dart]: @sethladd, @keertip, is this the best long-term address + format for dartdocs? It looks like we're using:
```
https://api.dartlang.org/apidocs/channels/stable/dartdoc-viewer.$libraryName/$className#$memberName
```

Comment (kasperpeulen) [inline lib/playground.dart]: @devoncarew I'm using `https://api.dartlang.org/apidocs/channels/stable/dartdoc-viewer/$libraryName.$className#id_$memberName`

For example: `https://api.dartlang.org/apidocs/channels/stable/dartdoc-viewer/dart:html.Range#id_comparePoint`

**Comment by devoncarew** [comment]
lgtm!

**Pull request merged**